integration/zero/testtools.clj

```diff
@@ -0,0 +1,112 @@
+                       :query-params {:uuid uuid}})]
+    (first (parse-json-string (:body response)))))
+
+(defn get-workloads
+  "Query v1 api for all workloads"
+  []
+  (let [auth-header (once/get-auth-header!)
+        response    (client/get (str server "/api/v1/workload")
+                      {:headers auth-header})]
+    (parse-json-string (:body response))))
+
+(def get-pending-workloads
+  "Query the v1 api for the workloads that has not been started"
+  (comp (partial remove :started) get-workloads))
+
+(def first-pending-workload
+  "Query the v1 api for first workload that has not been started"
+  (comp first get-pending-workloads))
+
+(defn first-pending-workload-or
+  "Query the v1 api for the first pending workload or evaluate the function
+  if no pending workload exists"
+  [alternative]
+  (let [workload (first-pending-workload)]
+    (if workload workload (alternative))))
+
+(defn create-workload
+  "Create workload defined by WORKLOAD"
+  [workload]
+  (let [auth-header (once/get-auth-header!)
+        payload     (json/write-str workload)
+        response    (client/post (str server "/api/v1/create")
+                      {:headers      auth-header
+                       :content-type :json
+                       :accept       :json
+                       :body         payload})]
+    (parse-json-string (:body response))))
+
+(defn start-workload
+  "Start processing WORKLOAD. WORKLOAD must be known to the server."
+  [workload]
+  (let [auth-header (once/get-auth-header!)
+        payload     (json/write-str [workload])
+        response    (client/post (str server "/api/v1/start")
+                      {:headers      auth-header
+                       :content-type :json
+                       :accept       :json
+                       :body         payload})]
+    (parse-json-string (:body response))))
+
+(defn start-wgs-workflow
+  "Submit a WGS Reprocessing workflow"
+  [workflow-description]
+  (let [auth-header (once/get-auth-header!)
+        payload     (json/write-str workflow-description :escape-slash false)
+        response    (client/post (str server "/api/v1/wgs")
+                      {:headers      auth-header
+                       :content-type :json
+                       :accept       :json
+                       :body         payload})]
+    (parse-json-string (:body response))))
+
+(defn exec-workload
+  "Create and start workload defined by WORKLOAD"
+  [workload]
+  (let [auth-header (once/get-auth-header!)
+        payload     (json/write-str workload)
+        response    (client/post (str server "/api/v1/exec")
+                      {:headers      auth-header
+                       :content-type :json
+                       :accept       :json
+                       :body         payload})]
+    (parse-json-string (:body response))))
+
+(def git-branch (delay (util/shell! "git" "branch" "--show-current")))
+(def git-email (delay (util/shell! "git" "config" "user.email")))
```

Comment:
Delay to speed up buffer reloading?

Reply:
Yeah I thought this was probably more hygienic to delay this to avoid shell calls during load